databricks/koalas/groupby.py

```diff
@@ -1401,6 +1401,67 @@ def ffill(self, limit=None):
 
     pad = ffill
 
+    def head(self, n=5):
+        """
+        Return first n rows of each group.
+
+        Returns
+        -------
+        DataFrame or Series
+
+        Examples
+        --------
+        >>> df = ks.DataFrame({'a': [1, 1, 1, 1, 2, 2, 2, 3, 3, 3],
+        ...                    'b': [2, 3, 1, 4, 6, 9, 8, 10, 7, 5],
+        ...                    'c': [3, 5, 2, 5, 1, 2, 6, 4, 3, 6]},
+        ...                   columns=['a', 'b', 'c'],
+        ...                   index=[7, 2, 4, 1, 3, 4, 9, 10, 5, 6])
+        >>> df
+            a   b  c
+        7   1   2  3
+        2   1   3  5
+        4   1   1  2
+        1   1   4  5
+        3   2   6  1
+        4   2   9  2
+        9   2   8  6
+        10  3  10  4
+        5   3   7  3
+        6   3   5  6
+
+        >>> df.groupby('a').head(2)
+            a   b  c
+        7   1   2  3
+        2   1   3  5
+        10  3  10  4
+        5   3   7  3
+        3   2   6  1
+        4   2   9  2
+
+        >>> df.groupby('a')['b'].head(2)
+        7      2
+        2      3
+        10    10
+        5      7
+        3      6
+        4      9
+        Name: b, dtype: int64
+        """
+        groupkeys = self._groupkeys
+        tmp_col = '__order__'
+        sdf = self._kdf._sdf.withColumn(tmp_col, monotonically_increasing_id())
+        name = self._agg_columns[0]._internal.data_columns[0]
+        window = Window.partitionBy([s._scol for s in groupkeys]).orderBy(F.col(tmp_col))
+        sdf = sdf.withColumn(tmp_col, F.row_number().over(window)).filter(F.col(tmp_col) <= n)
+        internal = _InternalFrame(
+            sdf=sdf,
+            index_map=self._kdf._internal.index_map,
+            column_scols=([scol_for(sdf, name)]
+                          if isinstance(self, SeriesGroupBy)
+                          else [scol_for(sdf, name) for name in self._kdf._internal.data_columns]))
```

Comment:
@HyukjinKwon for more your information related with https://github.com/databricks/koalas/pull/1050#discussion_r347227629, that `name` will be used when type of `self` is `SeriesGroupBy`, otherwise the `name` is re-generated by `[scol_for(sdf, name) for name in self._kdf._internal.data_columns]` here

Reply:
Let's move all related codes into where `name` variable exists. Looks difficult to follow.